sdk/trace/src/main/java/io/opentelemetry/sdk/trace/AttributesMap.java

```diff
@@ -9,40 +9,32 @@
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.common.AttributesBuilder;
 import java.util.Collections;
-import java.util.LinkedHashMap;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.function.BiConsumer;
 
-/**
- * A map with a fixed capacity that drops attributes when the map gets full.
- *
- * <p>Note: this doesn't implement the Map interface, but behaves very similarly to one.
- */
+/** A map with a fixed capacity that drops attributes when the map gets full. */
 @SuppressWarnings({"rawtypes", "unchecked"})
-final class AttributesMap implements Attributes {
-  private final Map<AttributeKey<?>, Object> data;
+final class AttributesMap extends HashMap<AttributeKey<?>, Object> implements Attributes {
```

Comment:
I definitely agree with not using the LinkedHashMap as the implementation any more. But, are we sure that a single allocation per span is worth introducing this antipattern? I know it's not a public class, but it feels like it would be very easy for us to start abusing this wide-open API, leading to much more confusing code.

Reply:
I'd generally say abuse anything if it's private :) The allocations are not paid by us, but by the apps, so happy to do what I can, to me it feels free. I'm also ok with reimplementing proto serialization if it helps for the same reason, it's in some sense an antipattern but it's all private code and helps apps.

Reply:
But yeah I think we can copy in putall to this class so in this case I think we can reduce some hackiness without losing performance so that is ideal will try it.

Reply:
Ok remember why we extend `Attributes` here - since in `toSpanData` we don't actually make an immutable copy if the span has ended already. So we'd need #2905 I think to be able to extend `HashMap` here - it seems nice to skip an allocation where we can though not the end of the world, but if it brings us more inline with JDK (and I guess best practice? I know I often forget about the `? super` deal but have a feeling it's generally what you're supposed to do) may as well?

Reply:
ok, Agrawal. I'll let this one through, but I'm watching you like a hawk for abuses of this antipattern. 👁️ 👁️ 🐦